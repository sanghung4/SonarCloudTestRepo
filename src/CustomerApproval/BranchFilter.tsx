import { Dispatch, ReactNode } from 'react';

import {
  Autocomplete,
  Box,
  CircularProgress,
  Grid,
  TextField,
  useScreenSize
} from '@dialexa/reece-component-library';
import { FN } from '@reece/global-types';
import { isUndefined } from 'lodash-es';
import { useTranslation } from 'react-i18next';

import {
  useBranchFilterAutocomplete,
  autocompleteOnChange
} from 'CustomerApproval/util/branchFilterUtil';
import { CoercedUser } from 'CustomerApproval/util/types';
import FilterActionButtons from 'common/TablePageLayout/FilterActionButtons';

type Props = {
  userList?: CoercedUser[];
  filterValue: string;
  setFilter: Dispatch<string>;
  submitFilter: FN;
  resetFilters: FN;
};

export default function BranchFilter(props: Props) {
  /**
   * Custom hooks
   */
  const { isSmallScreen } = useScreenSize();
  const { t } = useTranslation();
  const autocompleteOptions = useBranchFilterAutocomplete(props.userList);
  const undefinedUserList = isUndefined(props.userList);

  /**
   * Special component
   */
  function inputStartAdornment(startAdornment: ReactNode) {
    return (
      <>
        {undefinedUserList && (
          <Box ml={1}>
            <CircularProgress color="primary02.main" size={20} />
          </Box>
        )}
        {startAdornment}
      </>
    );
  }

  /**
   * Render
   */
  return (
    <Grid container spacing={2} alignItems="flex-end">
      <Grid item xs={isSmallScreen ? 12 : 4}>
        <Box marginBottom="-4px">
          <Autocomplete
            fullWidth
            value={undefinedUserList ? '' : props.filterValue || '-'}
            disableClearable
            disabled={undefinedUserList}
            id="branch-filter-autocomplete"
            options={autocompleteOptions}
            onChange={autocompleteOnChange(props.setFilter)}
            renderInput={(params) => (
              <TextField
                {...params}
                label={t('common.branch')}
                data-testid="branch-filter-input"
                id="branch-filter-input"
                InputProps={{
                  ...params.InputProps,
                  startAdornment: inputStartAdornment(
                    params.InputProps.startAdornment
                  )
                }}
              />
            )}
            data-testid="branch-filter"
          />
        </Box>
      </Grid>
      <FilterActionButtons
        dirty={props.filterValue !== t('common.all')}
        hideApplyOnMobile
        onViewResults={props.submitFilter}
        onReset={props.resetFilters}
      />
    </Grid>
  );
}
