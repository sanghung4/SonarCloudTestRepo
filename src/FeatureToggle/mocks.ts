import { Feature } from 'generated/graphql';

export const mockFeatures: Feature[] = [
  { id: '1', isEnabled: false, name: 'noooo' },
  { id: '2', isEnabled: true, name: 'fine' },
  { id: '3', isEnabled: true, name: 'test' }
];
