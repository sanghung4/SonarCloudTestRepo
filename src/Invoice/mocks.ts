import { GetInvoiceDocument } from 'generated/graphql';

export const successMincron = {
  request: {
    query: GetInvoiceDocument,
    variables: {
      accountId: '123456',
      invoiceNumber: '5460026'
    }
  },
  result: {
    data: {
      invoice: {
        invoiceNumber: '5460026',
        status: 'Open',
        terms: 'NET 30 DAYS',
        customerPo: '40251/13325',
        invoiceDate: '10/28/2020',
        dueDate: '10/28/2020',
        originalAmt: '',
        openBalance: '',
        age: '',
        jobNumber: 'AVLNEP7',
        jobName: 'NEP 7 ASHEVILLE',
        address: {
          streetLineOne: '911 BLUEBERRY WAY',
          streetLineTwo: '',
          streetLineThree: '',
          city: 'NORTHLAKE',
          state: 'TX',
          postalCode: '76226',
          country: ''
        },
        shipDate: '11/06/2020',
        deliveryMethod: 'Our Truck',
        subtotal: '3184.28',
        tax: '262.76',
        otherCharges: '0.0',
        paidToDate: '1000.00',
        invoiceItems: [
          {
            id: '1738433F4-91A7-4637-8C60-B4DF59A060C1',
            brand: 'Advanced Drainage Systems, Inc.',
            name: '120 x 7.5, 100 Sq Yd, Geosynthetic, Double, Photodegradable, Erosion Control Blanket',
            partNumber: '8X1125STRAWDN',
            mfr: '00S2TT',
            thumb: 'null',
            price: '43.59',
            qty: {
              quantityOrdered: 1,
              quantityShipped: 1
            }
          }
        ]
      }
    }
  }
};

export const successMincron_5448540 = {
  request: {
    query: GetInvoiceDocument,
    variables: {
      accountId: '91492',
      invoiceNumber: '5448540'
    }
  },
  result: {
    data: {
      invoice: {
        invoiceNumber: '5282411',
        status: 'Open',
        terms: 'NET 30 DAYS',
        customerPo: 'WILKESBORO JIM',
        invoiceDate: '10/06/2021',
        dueDate: '10/28/2020',
        originalAmt: '',
        openBalance: '',
        age: '',
        jobNumber: 'SHOP',
        jobName: 'SHOP',
        address: {
          streetLineOne: '911 BLUEBERRY WAY',
          streetLineTwo: '',
          streetLineThree: '',
          city: 'NORTHLAKE',
          state: 'TX',
          postalCode: '76226',
          country: ''
        },
        shipDate: '11/06/2021',
        deliveryMethod: 'Our Truck',
        subtotal: '2184.28',
        tax: '162.73',
        otherCharges: '0.0',
        paidToDate: '139.91',
        invoiceItems: {
          id: 'F02FD85F-4DCE-4EA7-8459-2DA7CFD689DC',
          brand: 'Certex USA, name:2 x 12, Nylon, Eye and Eye, Ply Sling',
          partNumber: 'SLING212,mfr:CX07-0126-12',
          thumb:
            'https://images.tradeservice.com/ProductImages/DIR100193/CERTEX_CX07-0126-12_LINE_SML.jpg',
          price: '40.00',
          qty: {
            quantityOrdered: 1,
            quantityShipped: 1
          }
        }
      }
    }
  }
};
