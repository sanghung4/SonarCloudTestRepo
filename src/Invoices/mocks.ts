import { subMonths } from 'date-fns';
import { Invoice, InvoicesDocument } from 'generated/graphql';
import { mockRow } from 'test-utils/mockTableInstance';
import { formatDate } from 'utils/dates';

export const success = {
  request: {
    query: InvoicesDocument,
    variables: {
      accountId: '123456',
      erpName: 'ECLIPSE'
    }
  },
  result: {
    data: {
      invoices: {
        bucketNinety: 0.0,
        bucketOneTwenty: 0.0,
        bucketSixty: 0.0,
        bucketThirty: 0.0,
        bucketFuture: 0.0,
        currentAmt: 35224.96,
        invoices: [
          {
            age: 'Future',
            customerPo: '109687',
            invoiceDate: formatDate(subMonths(new Date(), 1)),
            dueDate: '03/01/2022',
            invoiceNumber: 'S110227972.001',
            openBalance: '12.39',
            originalAmt: '12.39',
            status: 'Open',
            invoiceUrl: 'test.com',
            jobName: null,
            jobNumber: null
          },
          {
            age: 'Current',
            customerPo: '1096872',
            invoiceDate: formatDate(subMonths(new Date(), 1)),
            dueDate: '03/01/2022',
            invoiceNumber: 'S110227975.001',
            openBalance: '12.39',
            originalAmt: '12.39',
            status: 'Open',
            invoiceUrl: 'test.com',
            jobName: null,
            jobNumber: null
          },
          {
            age: 'Over120',
            customerPo: '109687',
            invoiceDate: formatDate(subMonths(new Date(), 1)),
            dueDate: '03/01/2022',
            invoiceNumber: 'S110220294.001',
            openBalance: '12.39',
            originalAmt: '12.39',
            status: 'Open',
            invoiceUrl: 'test.com',
            jobName: null,
            jobNumber: null
          },
          {
            age: '91-120',
            customerPo: '109687',
            invoiceDate: '03/01/2021',
            dueDate: '03/01/2022',
            invoiceNumber: 'S110222345.001',
            openBalance: '12.39',
            originalAmt: '12.39',
            status: 'Open',
            invoiceUrl: 'test.com',
            jobName: null,
            jobNumber: null
          },
          {
            age: 'Deposit',
            customerPo: '109687',
            invoiceDate: '03/01/2021',
            dueDate: '03/01/2022',
            invoiceNumber: 'S110220974.001',
            openBalance: '12.39',
            originalAmt: '12.39',
            status: 'Open',
            invoiceUrl: 'test.com',
            jobName: null,
            jobNumber: null
          },
          {
            age: '31-60',
            customerPo: '109687',
            invoiceDate: '03/01/2021',
            dueDate: '03/01/2022',
            invoiceNumber: 'S110227973.001',
            openBalance: '12.39',
            originalAmt: '12.39',
            status: 'Open',
            invoiceUrl: 'test.com',
            jobName: null,
            jobNumber: null
          }
        ],
        totalAmt: 35232.27,
        totalPastDue: 7.3085938
      }
    }
  }
};

export const successMincron = {
  request: {
    query: InvoicesDocument,
    variables: {
      accountId: '123456',
      erpName: 'MINCRON'
    }
  },
  result: {
    data: {
      invoices: {
        bucketNinety: 0.0,
        bucketOneTwenty: 0.0,
        bucketSixty: 0.0,
        bucketThirty: 0.0,
        bucketFuture: 0.0,
        currentAmt: 35224.96,
        invoices: [
          {
            age: 'Future',
            customerPo: '109687',
            invoiceDate: formatDate(subMonths(new Date(), 1)),
            dueDate: '03/01/2022',
            invoiceNumber: 'S110227972.001',
            openBalance: '12.39',
            originalAmt: '12.39',
            status: 'Open',
            invoiceUrl: 'test.com',
            jobNumber: '6566',
            jobName: 'NEP 7 ASHVELLE'
          },
          {
            age: 'Current',
            customerPo: '109687',
            invoiceDate: formatDate(subMonths(new Date(), 1)),
            dueDate: '03/01/2022',
            invoiceNumber: 'S110227975.001',
            openBalance: '12.39',
            originalAmt: '12.39',
            status: 'Open',
            invoiceUrl: 'test.com',
            jobNumber: '56787',
            jobName: 'WALTER HORN'
          },
          {
            age: 'Over120',
            customerPo: '109687',
            invoiceDate: formatDate(subMonths(new Date(), 1)),
            dueDate: '03/01/2022',
            invoiceNumber: 'S110220294.001',
            openBalance: '12.39',
            originalAmt: '12.39',
            status: 'Closed',
            invoiceUrl: 'test.com',
            jobNumber: 'SHOP',
            jobName: 'SEQUAYESH TR'
          },
          {
            age: '91-120',
            customerPo: '109687',
            invoiceDate: formatDate(subMonths(new Date(), 1)),
            dueDate: '03/01/2022',
            invoiceNumber: 'S110222345.001',
            openBalance: '12.39',
            originalAmt: '12.39',
            status: 'Open',
            invoiceUrl: 'test.com',
            jobNumber: 'RETREAT',
            jobName: 'RETREAT AT ARDEN'
          },
          {
            age: 'Deposit',
            customerPo: '109687',
            invoiceDate: formatDate(subMonths(new Date(), 1)),
            dueDate: '03/01/2022',
            invoiceNumber: 'S110220974.001',
            openBalance: '12.39',
            originalAmt: '12.39',
            status: 'Closed',
            invoiceUrl: 'test.com',
            jobNumber: 'AVLNEP7',
            jobName: 'WALTER HORN'
          },
          {
            age: '31-60',
            customerPo: '109687',
            invoiceDate: '03/01/2021',
            dueDate: '03/01/2022',
            invoiceNumber: 'S110227973.001',
            openBalance: '12.39',
            originalAmt: '12.39',
            status: 'Open',
            invoiceUrl: 'test.com',
            jobNumber: 'AIKENON',
            jobName: 'AIKEN AVL'
          }
        ],
        totalAmt: 35232.27,
        totalPastDue: 7.3085938
      }
    }
  }
};

export const mockTestTableMemo = {
  invoices: {
    bucketNinety: 0.0,
    bucketOneTwenty: 0.0,
    bucketSixty: 0.0,
    bucketThirty: 0.0,
    bucketFuture: 0.0,
    currentAmt: 35224.96,
    invoices: [
      {
        age: 'Future',
        customerPo: '109687',
        invoiceDate: '05/02/2022',
        dueDate: '03/01/2022',
        invoiceNumber: 'S110227972.001',
        openBalance: '12.39',
        originalAmt: '12.39',
        status: 'Open',
        invoiceUrl: 'test.com',
        jobNumber: '6566',
        jobName: 'NEP 7 ASHVELLE'
      },
      {
        age: 'Current',
        customerPo: '109687',
        invoiceDate: '03/01/2021',
        dueDate: '03/01/2022',
        invoiceNumber: 'S110227975.001',
        openBalance: '12.39',
        originalAmt: '12.39',
        status: 'Open',
        invoiceUrl: 'test.com',
        jobNumber: '56787',
        jobName: 'WALTER HORN'
      },
      {
        age: 'Over120',
        customerPo: '109687',
        invoiceDate: '03/01/2021',
        dueDate: '03/01/2022',
        invoiceNumber: 'S110220294.001',
        openBalance: '12.39',
        originalAmt: '12.39',
        status: 'Open',
        invoiceUrl: 'test.com',
        jobNumber: 'SHOP',
        jobName: 'SEQUAYESH TR'
      },
      {
        age: '91-120',
        customerPo: '109687',
        invoiceDate: '03/01/2021',
        dueDate: '03/01/2022',
        invoiceNumber: 'S110222345.001',
        openBalance: '12.39',
        originalAmt: '12.39',
        status: 'Open',
        invoiceUrl: 'test.com',
        jobNumber: 'RETREAT',
        jobName: 'RETREAT AT ARDEN'
      },
      {
        age: 'Deposit',
        customerPo: '109687',
        invoiceDate: '03/01/2021',
        dueDate: '03/01/2022',
        invoiceNumber: 'S110220974.001',
        openBalance: '12.39',
        originalAmt: '12.39',
        status: 'Open',
        invoiceUrl: 'test.com',
        jobNumber: 'AVLNEP7',
        jobName: 'WALTER HORN'
      },
      {
        age: '31-60',
        customerPo: '109687',
        invoiceDate: '03/01/2021',
        dueDate: '03/01/2022',
        invoiceNumber: 'S110227973.001',
        openBalance: '12.39',
        originalAmt: '12.39',
        status: 'Open',
        invoiceUrl: 'test.com',
        jobNumber: 'AIKENON',
        jobName: 'AIKEN AVL'
      }
    ],
    totalAmt: 35232.27,
    totalPastDue: 7.3085938
  }
};

const rowToMock = success.result.data.invoices.invoices[0];
export const mockInvoiceRow = mockRow<Invoice>(rowToMock, rowToMock, {});
