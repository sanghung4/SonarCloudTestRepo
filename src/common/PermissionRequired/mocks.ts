import { UserContextType } from 'AuthProvider';
import { Permission } from 'common/PermissionRequired';

export const authConfigNoPerm = {
  authState: {
    isAuthenticated: true
  },
  profile: {
    isEmployee: true,
    isVerified: true
  } as UserContextType['profile'],
  activeFeatures: ['LISTS']
};
export const authConfig = {
  authState: {
    isAuthenticated: true
  },
  profile: {
    userId: 'test',
    permissions: [Permission.EDIT_LIST, Permission.APPROVE_CART],
    isEmployee: true,
    isVerified: true
  } as UserContextType['profile'],
  activeFeatures: ['LISTS']
};
