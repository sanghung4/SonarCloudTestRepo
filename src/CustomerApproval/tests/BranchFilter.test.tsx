import BranchFilter from 'CustomerApproval/BranchFilter';
import { mockCoercedUsers } from 'CustomerApproval/tests/mocks';
import { setBreakpoint } from 'test-utils/mockMediaQuery';
import { render } from 'test-utils/TestWrapper';

/**
 * Mock props
 */
const mockDefaultProps = {
  filterValue: '',
  setFilter: jest.fn(),
  submitFilter: jest.fn(),
  resetFilters: jest.fn()
};

/**
 * Test
 */
describe('CustomerApproval - BranchFilter', () => {
  it('Should match snaptshot with no userList on desktop', () => {
    setBreakpoint('desktop');
    const { container } = render(<BranchFilter {...mockDefaultProps} />);
    expect(container).toMatchSnapshot();
  });

  it('Should match snaptshot with no userList on mobile', () => {
    setBreakpoint('mobile');
    const { container } = render(<BranchFilter {...mockDefaultProps} />);
    expect(container).toMatchSnapshot();
  });

  it('Should match snaptshot with userList', () => {
    setBreakpoint('desktop');
    const { container } = render(
      <BranchFilter {...mockDefaultProps} userList={mockCoercedUsers} />
    );
    expect(container).toMatchSnapshot();
  });
});
