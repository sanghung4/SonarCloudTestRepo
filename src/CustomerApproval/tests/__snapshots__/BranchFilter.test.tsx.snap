// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomerApproval - BranchFilter Should match snaptshot with no userList on desktop 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-jhdc2n-MuiGrid-root"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-1rqx6ho-MuiGrid-root"
    >
      <div
        class="MuiBox-root css-1i3vb4c"
      >
        <div
          aria-expanded="false"
          class="MuiAutocomplete-root MuiAutocomplete-fullWidth MuiAutocomplete-hasPopupIcon css-eozd4h-MuiAutocomplete-root"
          data-testid="branch-filter"
          role="combobox"
        >
          <div
            class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-1vhyr2u-MuiFormControl-root-MuiTextField-root"
            data-testid="branch-filter-input"
          >
            <label
              class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary Mui-disabled css-1411gyj-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="true"
              for="branch-filter-autocomplete"
              id="branch-filter-autocomplete-label"
            >
              Branch
            </label>
            <div
              class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary Mui-disabled MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedStart MuiInputBase-adornedEnd MuiAutocomplete-inputRoot css-dndmr2-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <div
                class="MuiBox-root css-1isemmb"
              >
                <div
                  class="MuiBox-root css-hyxlzm"
                >
                  <span
                    aria-valuenow="100"
                    class="MuiCircularProgress-root MuiCircularProgress-determinate MuiCircularProgress-colorPrimary css-1wyqyef-MuiCircularProgress-root"
                    role="progressbar"
                    style="width: 20px; height: 20px; transform: rotate(-90deg);"
                  >
                    <svg
                      class="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
                      viewBox="22 22 44 44"
                    >
                      <circle
                        class="MuiCircularProgress-circle MuiCircularProgress-circleDeterminate css-oxts8u-MuiCircularProgress-circle"
                        cx="44"
                        cy="44"
                        fill="none"
                        r="19.5"
                        stroke-width="5"
                        style="stroke-dasharray: 122.522; stroke-dashoffset: 0.000px;"
                      />
                    </svg>
                  </span>
                  <span
                    class="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary css-1gse1c1-MuiCircularProgress-root"
                    role="progressbar"
                    style="width: 20px; height: 20px;"
                  >
                    <svg
                      class="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
                      viewBox="22 22 44 44"
                    >
                      <circle
                        class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle"
                        cx="44"
                        cy="44"
                        fill="none"
                        r="19.5"
                        stroke-width="5"
                      />
                    </svg>
                  </span>
                </div>
              </div>
              <input
                aria-autocomplete="list"
                aria-invalid="false"
                autocapitalize="none"
                autocomplete="off"
                class="MuiOutlinedInput-input MuiInputBase-input Mui-disabled MuiInputBase-inputAdornedStart MuiInputBase-inputAdornedEnd MuiAutocomplete-input MuiAutocomplete-inputFocused css-fxwsjr-MuiInputBase-input-MuiOutlinedInput-input"
                disabled=""
                id="branch-filter-autocomplete"
                spellcheck="false"
                type="text"
                value=""
              />
              <div
                class="MuiAutocomplete-endAdornment css-1q60rmi-MuiAutocomplete-endAdornment"
              >
                <button
                  aria-label="Open"
                  class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-sizeMedium MuiAutocomplete-popupIndicator css-qzbt6i-MuiButtonBase-root-MuiIconButton-root-MuiAutocomplete-popupIndicator"
                  disabled=""
                  tabindex="-1"
                  title="Open"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="ArrowDropDownIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M7 10l5 5 5-5z"
                    />
                  </svg>
                </button>
              </div>
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-m1dym3-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-1z7n62"
                >
                  <span>
                    Branch
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-auto css-11vcf2t-MuiGrid-root"
    >
      <button
        class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-disableElevation MuiButton-fullWidth MuiButtonBase-root  css-190vp1s-MuiButtonBase-root-MuiButton-root"
        data-testid="branch-filter-set-filter"
        tabindex="0"
        type="button"
      >
        View Results
      </button>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-auto css-11vcf2t-MuiGrid-root"
    >
      <button
        class="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-disableElevation MuiButton-fullWidth MuiButtonBase-root  css-yxavb6-MuiButtonBase-root-MuiButton-root"
        data-testid="reset-button"
        tabindex="0"
        type="button"
      >
        Reset
      </button>
    </div>
  </div>
</div>
`;

exports[`CustomerApproval - BranchFilter Should match snaptshot with no userList on mobile 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-jhdc2n-MuiGrid-root"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-49904w-MuiGrid-root"
    >
      <div
        class="MuiBox-root css-1i3vb4c"
      >
        <div
          aria-expanded="false"
          class="MuiAutocomplete-root MuiAutocomplete-fullWidth MuiAutocomplete-hasPopupIcon css-eozd4h-MuiAutocomplete-root"
          data-testid="branch-filter"
          role="combobox"
        >
          <div
            class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-1vhyr2u-MuiFormControl-root-MuiTextField-root"
            data-testid="branch-filter-input"
          >
            <label
              class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary Mui-disabled css-1411gyj-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="true"
              for="branch-filter-autocomplete"
              id="branch-filter-autocomplete-label"
            >
              Branch
            </label>
            <div
              class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary Mui-disabled MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedStart MuiInputBase-adornedEnd MuiAutocomplete-inputRoot css-dndmr2-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <div
                class="MuiBox-root css-1isemmb"
              >
                <div
                  class="MuiBox-root css-hyxlzm"
                >
                  <span
                    aria-valuenow="100"
                    class="MuiCircularProgress-root MuiCircularProgress-determinate MuiCircularProgress-colorPrimary css-1wyqyef-MuiCircularProgress-root"
                    role="progressbar"
                    style="width: 20px; height: 20px; transform: rotate(-90deg);"
                  >
                    <svg
                      class="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
                      viewBox="22 22 44 44"
                    >
                      <circle
                        class="MuiCircularProgress-circle MuiCircularProgress-circleDeterminate css-oxts8u-MuiCircularProgress-circle"
                        cx="44"
                        cy="44"
                        fill="none"
                        r="19.5"
                        stroke-width="5"
                        style="stroke-dasharray: 122.522; stroke-dashoffset: 0.000px;"
                      />
                    </svg>
                  </span>
                  <span
                    class="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary css-1gse1c1-MuiCircularProgress-root"
                    role="progressbar"
                    style="width: 20px; height: 20px;"
                  >
                    <svg
                      class="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
                      viewBox="22 22 44 44"
                    >
                      <circle
                        class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle"
                        cx="44"
                        cy="44"
                        fill="none"
                        r="19.5"
                        stroke-width="5"
                      />
                    </svg>
                  </span>
                </div>
              </div>
              <input
                aria-autocomplete="list"
                aria-invalid="false"
                autocapitalize="none"
                autocomplete="off"
                class="MuiOutlinedInput-input MuiInputBase-input Mui-disabled MuiInputBase-inputAdornedStart MuiInputBase-inputAdornedEnd MuiAutocomplete-input MuiAutocomplete-inputFocused css-fxwsjr-MuiInputBase-input-MuiOutlinedInput-input"
                disabled=""
                id="branch-filter-autocomplete"
                spellcheck="false"
                type="text"
                value=""
              />
              <div
                class="MuiAutocomplete-endAdornment css-1q60rmi-MuiAutocomplete-endAdornment"
              >
                <button
                  aria-label="Open"
                  class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-sizeMedium MuiAutocomplete-popupIndicator css-qzbt6i-MuiButtonBase-root-MuiIconButton-root-MuiAutocomplete-popupIndicator"
                  disabled=""
                  tabindex="-1"
                  title="Open"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="ArrowDropDownIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M7 10l5 5 5-5z"
                    />
                  </svg>
                </button>
              </div>
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-m1dym3-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-1z7n62"
                >
                  <span>
                    Branch
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-auto css-11vcf2t-MuiGrid-root"
    >
      <button
        class="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-disableElevation MuiButton-fullWidth MuiButtonBase-root  css-1216ci0-MuiButtonBase-root-MuiButton-root"
        data-testid="reset-button"
        tabindex="0"
        type="button"
      >
        Reset
      </button>
    </div>
  </div>
</div>
`;

exports[`CustomerApproval - BranchFilter Should match snaptshot with userList 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-jhdc2n-MuiGrid-root"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-1rqx6ho-MuiGrid-root"
    >
      <div
        class="MuiBox-root css-1i3vb4c"
      >
        <div
          aria-expanded="false"
          class="MuiAutocomplete-root MuiAutocomplete-fullWidth MuiAutocomplete-hasPopupIcon css-eozd4h-MuiAutocomplete-root"
          data-testid="branch-filter"
          role="combobox"
        >
          <div
            class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-1vhyr2u-MuiFormControl-root-MuiTextField-root"
            data-testid="branch-filter-input"
          >
            <label
              class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary MuiFormLabel-filled css-1411gyj-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="true"
              for="branch-filter-autocomplete"
              id="branch-filter-autocomplete-label"
            >
              Branch
            </label>
            <div
              class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedStart MuiInputBase-adornedEnd MuiAutocomplete-inputRoot css-dndmr2-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <input
                aria-autocomplete="list"
                aria-invalid="false"
                autocapitalize="none"
                autocomplete="off"
                class="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputAdornedStart MuiInputBase-inputAdornedEnd MuiAutocomplete-input MuiAutocomplete-inputFocused css-fxwsjr-MuiInputBase-input-MuiOutlinedInput-input"
                id="branch-filter-autocomplete"
                spellcheck="false"
                type="text"
                value="-"
              />
              <div
                class="MuiAutocomplete-endAdornment css-1q60rmi-MuiAutocomplete-endAdornment"
              >
                <button
                  aria-label="Open"
                  class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium MuiAutocomplete-popupIndicator css-qzbt6i-MuiButtonBase-root-MuiIconButton-root-MuiAutocomplete-popupIndicator"
                  tabindex="-1"
                  title="Open"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="ArrowDropDownIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M7 10l5 5 5-5z"
                    />
                  </svg>
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
              </div>
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-m1dym3-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-1z7n62"
                >
                  <span>
                    Branch
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-auto css-11vcf2t-MuiGrid-root"
    >
      <button
        class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-disableElevation MuiButton-fullWidth MuiButtonBase-root  css-190vp1s-MuiButtonBase-root-MuiButton-root"
        data-testid="branch-filter-set-filter"
        tabindex="0"
        type="button"
      >
        View Results
      </button>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-auto css-11vcf2t-MuiGrid-root"
    >
      <button
        class="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-disableElevation MuiButton-fullWidth MuiButtonBase-root  css-yxavb6-MuiButtonBase-root-MuiButton-root"
        data-testid="reset-button"
        tabindex="0"
        type="button"
      >
        Reset
      </button>
    </div>
  </div>
</div>
`;
